src/lib/rnp.c

```diff
@@ -1371,8 +1371,48 @@ rnp_armor_stream(
         return RNP_ERROR_WRITE;
     }
 
-    result = is_armor ? rnp_armor_source(&src, &dst, (pgp_armored_msg_t) data_type) :
-                        rnp_dearmor_source(&src, &dst);
+    if ((msgt = (pgp_armored_msg_t) ctx->armortype) == PGP_ARMORED_UNKNOWN) {
+        msgt = rnp_armor_guess_type(&src);
+    }
+
+    result = rnp_armor_source(&src, &dst, msgt);
+
+    if (result != RNP_SUCCESS) {
+        RNP_LOG("error code 0x%x", result);
+        return result;
+    }
+
+    src_close(&src);
+    dst_close(&dst, result != RNP_SUCCESS);
+    return result;
+}
+
+rnp_result_t
+rnp_dearmor_stream(rnp_ctx_t *ctx, const char *in, const char *out)
+{
+    pgp_source_t src;
+    pgp_dest_t   dst;
+    rnp_result_t result;
+    bool         is_stdin;
+    bool         is_stdout;
+
+    is_stdin = !in || (strlen(in) == 0) || (strcmp(in, "-") == 0);
```

Comment:
that's simply code  duplication. let's see how it looks after rebase when the other PR is integrated

Reply:
After merging other PR the duplication will be only in `rnp_initialize_input`/`rnp_initialize_input` calls. Actually this will be duplicated in encrypt/sign/decrypt calls as well. Probably I'll wrap it to something better later on, within implementation of #456

Reply:
...also the main reason for making two functions is the fact that logically armoring and dearmoring are different operations. First one generates data, second - parses. First one may have additional parameters (for instance, we may wish later on to add other line length, line ending, make it raw base64 without the headers, and so on). It's like mixing together encryption and decryption and writing inside `if (decrypt) {...do decryption...} else {...do encryption...}`

Reply:
I don't think that's good enough reason. In fact mixing decryption/encryption is good example - interfaces often use same method to encrypt and decrypt (CipherInit(), CipherUpdate(), CiperFinalize()).
I'm not saying two methods are not good, I'm saying code duplication must be avoided.

Reply:
While we are talking about raw symmetric encryption/decryption - then yeah, this may be done in single function call (however, it is not like this - many libraries have different calls for encrypt and decrypto). However raw encryption/decryption does not parse input data.

For PGP decryption/encryption (and, armoring/dearmoring) situation is opposite - a lot of efforts put into data parsing, so it's quite logical to separate those calls. Anyway, it has just few lines of code duplication.